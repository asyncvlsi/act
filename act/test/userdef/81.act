deftype foo(bool x)
{
  methods {
    function testme (pint x) : pint
    {
      chp {
        self := x + 1
      }
    }
  }
}

defproc test()
{
   foo f;
   int x;
 
   chp {
     x := f.testme(3);
     x := f.testme(x)
   }
}

test t;
