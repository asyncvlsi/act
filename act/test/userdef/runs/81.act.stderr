ERROR: File `81.act', line 20, col 13
	Typechecking failed on expression!
	User macro `testme': arg #0 has an incompatible type
